packages/interflex/src/interflex.ts

```diff
@@ -0,0 +1,139 @@
+const PLACEHOLDER = /\{(\w+)\}/g;
+const FRAGMENT = /<(\w+)>(.*?)<\/\1>/g;
+const DEFAULT_LOCALE_DATA: LocaleData = {
+  localizedName: '',
+  strings: {},
+};
+
+type LocaleData = {
+  localizedName: string;
+  strings: Record<string, string>;
+};
+
+type LocaleDictionary<T extends string> = { [key in T]: LocaleData };
+
+type Params = {
+  [key: string]:
+    | null
+    | undefined
+    | number
+    | string
+    | boolean
+    | ((s: string) => unknown);
+};
+
+export default class Interflex<TLocale extends string> {
+  private _locales: LocaleDictionary<TLocale>;
+  private _locale: TLocale;
+
+  constructor(defaultLocale: TLocale) {
+    this._locales = {} as LocaleDictionary<TLocale>;
+    this._locale = defaultLocale;
+  }
+
+  /**
+   * retrieves the current locale data.
+   */
+  get localeData() {
+    return this._locales[this._locale];
+  }
+
+  /**
+   * adds locale data into the dictionary.
+   * @param locale
+   * @param data
+   */
+  public addLocaleData = (locale: TLocale, data: LocaleData): void => {
+    this._locales[locale] = data;
+  };
+
+  /**
+   * adds more translation strings to a specific locale.
+   * @param locale
+   * @param strings
+   */
+  public addLocaleStrings = (
```

Comment:
Is this purely for testing? Because I assume that all strings would be defined statically, not at runtime.